### PR TITLE
Validate create proposal form

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**
using [Formik](https://github.com/jaredpalmer/formik).

### CODE REVIEW

**Comment by dev-matan-tsuberi** [comment]
[Consistent form validation](https://trello.com/c/QoFi2IHg/125-consistent-form-validation)

**Comment by dev-matan-tsuberi** [comment]
["Create Proposal" form, URL is not validated](https://trello.com/c/0ToKXm10/223-create-proposal-form-url-is-not-validated)

**Comment by dev-matan-tsuberi** [comment]
["Create Proposal" form: Reputation reward & ETH reward are not validated to be positive numbers](https://trello.com/c/LE7oTNYZ/224-create-proposal-form-reputation-reward-eth-reward-are-not-validated-to-be-positive-numbers)

File: package.json [hunk 1]
```diff
@@ -60,6 +60,7 @@
   },
   "dependencies": {
     "@daostack/arc.js": "0.0.0-alpha.58",
+    "@types/prop-types": "^15.5.3",
```
Comment (tibetsprague) [inline package.json]: should be a dev-dependency

File: src/components/CreateProposal/CreateProposalContainer.tsx [hunk 2]
```diff
@@ -159,125 +117,189 @@ class CreateProposalContainer extends React.Component<IProps, IState> {
...
+              description,
+              ethReward,
+              externalTokenReward,
+              nativeTokenReward,
+              reputationReward,
+              title
+            } = values;
+            const errors: any = {};
+
+            const require = (name: string) => {
```
Comment (tibetsprague) [inline src/components/CreateProposal/CreateProposalContainer.tsx]: seems really weird that we have to define require here. so we have to do this for every form we want to validate?

Comment (dev-matan-tsuberi) [reply src/components/CreateProposal/CreateProposalContainer.tsx]: just for convenience really... Formik doesn't provide built in validation rules. We can use HTML5's `required` but the error message would look different than the others. 
If it gets used more than once we'll extract it to a separate module.

**Pull request merged**